### PR TITLE
TMEDIA-519 - Log In Redirect Updates

### PR DESCRIPTION
**Pull request by matthewroach**
## Description
Move duplicated logic to a useLogin hook
Implement the logic to look for redirect query param

## Jira Ticket
- [TMEDIA-519](https://arcpublishing.atlassian.net/browse/TMEDIA-519)

## Acceptance Criteria
* update Login Block to look at redirect query param
* on successful login, redirect the browser to that URL

```
Logged out state:
// redirectQuery
// redirectToPreviousPage && referer
// redirectURL

Logged in state:
// redirectQuery
// loggedInPageLocation
```

## Test Steps

1. Checkout this branch `git checkout TMEDIA-519-login-redirects`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Validate login works as currently does
    * Login will redirect to the custom field setting or referrer
    * If you visit login page after logged in you are redirect to the logged in custom field values
4. Validate when you visit the login page with a redirect query param - http://localhost/account/login/?_website=the-gazette&redirect=http://localhost/homepage/?_website=the-gazette
    * You are redirected to the URL in the param after successful login
    * You are redirected to the URL in the param if you visit the page and are already logged i
 

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/identity-block/components/useLogin.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { isServerSide } from '@wpmedia/engine-theme-sdk/dist/cjs/utils';
```
Comment (JackHowa) [inline blocks/identity-block/components/useLogin.jsx]: is there a reason we're not using `import { isServerSide } from '@wpmedia/engine-theme-sdk';
`?

File: blocks/identity-block/features/social-sign-on/default.jsx [hunk 2]
```diff
@@ -58,7 +40,7 @@ const SocialSignOnBlock = ({ customFields }) => {
     <section className="xpmedia-subs-social-sign-on">
       <SocialSignOn
         onError={() => { setError(phrases.t('identity-block.login-form-error')); }}
-        redirectURL={redirectURL}
+        redirectURL={loginRedirect}
```
Comment (JackHowa) [inline blocks/identity-block/features/social-sign-on/default.jsx:43]: this seems to have been renamed. No big deal just wanted to call it out

**Review by JackHowa — DISMISSED** [review]
Functionality works as described in my testing. Had one comment about the import but otherwise looks good

File: blocks/identity-block/components/useLogin.jsx (hunk 1 above)
Comment (matthewroach) [reply blocks/identity-block/components/useLogin.jsx]: Ah my bad, must of done auto import

File: blocks/identity-block/features/social-sign-on/default.jsx (hunk 2 above)
Comment (matthewroach) [reply blocks/identity-block/features/social-sign-on/default.jsx:43]: Yeah just a renamed variable to make it clear

File: blocks/identity-block/components/useLogin.jsx (hunk 1 above)
Comment (matthewroach) [reply blocks/identity-block/components/useLogin.jsx]: Fixed this now @JackHowa

**Review by JackHowa — APPROVED** [review]
looks good

**Pull request merged**